qemu/tests/dump_guest_memory.py

```diff
@@ -0,0 +1,195 @@
+        :type key: str
+        :param val: expect result
+        :type val: str or None(if check_item_in_pair=False)
+        :param check_item_in_pair: If expect result is dict or str
+        :type check_item_in_pair: bool. Means expect result is dict or str.
+
+        :return check result
+        :rtype: bool
+        """
+        if key in dic and not check_item_in_pair:
+            return True
+        elif key in dic and val == dic[key]:
+            return True
+        else:
+            for value in dic.values():
+                if isinstance(value, dict):
+                    if _check_dict(value, key, val, check_item_in_pair):
+                        return True
+                elif isinstance(value, list):
+                    if check_list(value, key, val, check_item_in_pair):
+                        return True
+                elif value != '' and not check_item_in_pair:
+                    if key in str(value):
+                        return True
+            return False
+
+    def check_result(qmp_o, expect_o=None):
+        """
+        Check test result with difference way according to result_check.
+        result_check = equal, expect_o should equal to qmp_o.
+        result_check = contain, expect_o should be contained in qmp_o
+
+        :param qmp_o: output from qmp_cmd.
+        :type qmp_o: list
+        :param expect_o: the expect result.
+        :type expect_o: list
+        """
+        logging.info("Expect result is %s" % expect_o)
+        logging.info("Actual result that get from qmp_cmd is %s" % qmp_o)
+        result = None
+        if result_check == "equal":
+            if not operator.eq(qmp_o, expect_o):
+                test.fail("QMP output does not equal to the expect result.\n "
+                          "Expect result: '%s'\n"
+                          "Actual result: '%s'" % (expect_o, qmp_o))
+        elif result_check == "contain":
+            if len(expect_o) == 0:
+                result = True
+            elif isinstance(expect_o, dict):
+                for key, val in expect_o.items():
+                    result = check_list(qmp_o, key, val)
+        return result
+
+    def execute_qmp_cmd(qmp_cmd, expect_result):
+        """
+        Execute qmp command and check if result as expect
+
+        :param qmp_cmd: qmp command
+        :type qmp_cmd: str
+        :param expect_result: expect result of qmp command
+        :type expect_result: str
+        """
+        # qmp command
+        try:
+            # Testing command
+            logging.info("Run qmp command '%s'.", qmp_cmd)
+            qmp_o = qmp_port.send_args_cmd(qmp_cmd)
+            if not isinstance(qmp_o, list):
+                qmp_o = [qmp_o]
+            logging.debug("QMP command: '%s' \n Output: '%s'", qmp_cmd, qmp_o)
+        except Exception as err:
+            qmp_o = err.data
+            logging.info(err)
+
+        if result_check == "equal" or result_check == "contain":
```

Comment:
Here could be `if expect_result:` and Line 114-115 could be removed. `result_check` has no other condition anyway.